src/control/server/ctl_storage_rpc_test.go

```diff
@@ -213,9 +297,38 @@ func TestServer_CtlSvc_StorageScan(t *testing.T) {
 					WithBdevDeviceList(storage.MockNvmeController().PciAddr),
 			)
 
+			ioserverCount := len(tc.drpcResps)
+			if ioserverCount == 0 {
+				ioserverCount = 1
+			}
+			svc := newTestMgmtSvcMulti(t, log, ioserverCount, false)
+			for i, srv := range svc.harness.instances {
+				cfg := new(mockDrpcClientConfig)
+				if tc.junkResp {
+					cfg.setSendMsgResponse(drpc.Status_SUCCESS, makeBadBytes(42), nil)
+				} else if len(tc.drpcResps) > i {
+					for _, mock := range tc.drpcResps[i] {
+						cfg.setSendMsgResponseList(t, mock)
+					}
+				}
+				srv.setDrpcClient(newMockDrpcClient(cfg))
+			}
+			if !tc.ioStarted {
+				for _, srv := range svc.harness.instances {
+					srv.ready.SetFalse()
+				}
+			}
+
 			// test for both empty and default config cases
 			for configIdx, config := range []*Configuration{defaultWithNvme, emptyCfg} {
+				if configIdx == 1 && tc.req.ConfigDevicesOnly {
+					continue
+				}
+
 				cs := mockControlService(t, log, config, tc.bmbc, tc.smbc, nil)
+				if configIdx == 0 {
```

Comment:
These configIdx-based checks are a little confusing to me. Is there a reason we wouldn't want to use the mock all the time? Or care only about req.ConfigDevicesOnly regardless of which config we're looking at?

Reply:
tidied this up a little, now passing in Health/Meta (the interesting portion) in that request , so it's a bit more explicit. if it's an empty config then we don't want to override the harness with active instances (the instance list should be empty)

Reply:
refactored further to remove configIdx-based checks